src/orthonormal_bases.jl

```diff
@@ -343,3 +343,66 @@ function get_basis(M::Manifold, x, B::ProjectedOrthonormalBasis{:gram_schmidt,
     @warn "get_basis with bases $(typeof(B)) only found $(K) orthonormal basis vectors, but manifold dimension is $(dim)."
     return PrecomputedOrthonormalBasis(Ξ)
 end
+
+function _show_basis_vector(io::IO, v; pre = "", head = "")
+    sx = sprint(show, "text/plain", v, context = io, sizehint = 0)
+    sx = replace(sx, '\n' => "\n$(pre)")
+    print(io, head, pre, sx)
+end
+
+function _show_basis_vector_range(io::IO, vs, range; pre = "", sym = "E")
+    for i in range
+        _show_basis_vector(io, vs[i]; pre = pre, head = "\n$(sym)$(i) =\n")
+    end
+    return nothing
+end
+
+function show(io::IO, ::ArbitraryOrthonormalBasis{F}) where {F}
+    print(io, "ArbitraryOrthonormalBasis($(F))")
+end
+function show(io::IO, ::ProjectedOrthonormalBasis{method,F}) where {method,F}
+    print(io, "ProjectedOrthonormalBasis($(repr(method)), $(F))")
+end
+function show(io::IO, mime::MIME"text/plain", onb::DiagonalizingOrthonormalBasis)
+    println(
+        io,
+        "DiagonalizingOrthonormalBasis with coordinates in $(number_system(onb)) and 0 curvature in direction:",
+    )
+    sk = sprint(show, "text/plain", onb.v, context = io, sizehint = 0)
+    sk = replace(sk, '\n' => "\n ")
+    print(io, sk)
+end
+function show(io::IO, mime::MIME"text/plain", onb::PrecomputedOrthonormalBasis)
```

Comment:
Comparing this one with the next (Diagonalizing) I noticed that the next one prints an additional `:`  as well as `"Basis vectors:"´. Is that intended that way? Further the then-following lines are quite similar, they could also be put in a `_show_basis_vector_range_shortened` (or such) function that takes `io, vectors, limit=4,pre="",sym="")` ? Then one can even easily split this limit in the middle to obtain the current split idea.

Reply:
> Comparing this one with the next (Diagonalizing) I noticed that the next one prints an additional `:` as well as `"Basis vectors:"´. Is that intended that way? `

Yes, the second needs the `Basis vectors:` to clearly differentiate them from eigenvalues. If you think it would be better, I can also add `Basis vectors:` to the first to unify them. Good idea on the shared utility function.

Reply:
A, I missed the `Eigenvalues`, no than that difference is fine.